### ISSUE TITLE
Empty Response is not always empty.

### ISSUE THREAD

**Issue by deAtog** [symfony/symfony#39147]
**Symfony version(s) affected**: 5.1.8

**Description**  
Returning an empty response results sometimes results in having a Content-Length of 1 instead of 0.

**How to reproduce**  
Define a route using the following:
```php
   /**
    * @Route("/data")
    */
   public function getData() {
       return new Response();
   }
```
Retrieve response via the above route using ajax using various content-types. With Chrome and Firefox, using a context-type of 'text/plain' results in a response containing a single space and a Content-Length of 1 instead of 0. If a content-type of 'application/json' is specified, the response has a Content-Length of 0. I do not know what causes this issue and have not been able to reproduce the issue outside the browser. This issue did not occur in Symfony 4.1

**Possible Solution**  
I have no idea what causes this issue, so I cannot recommend a solution.

**Additional context**
Modifying the above code to set the content-type of the response based on the content-type of the request did not correct this issue. Another user has observed this issue via curl using `curl -s http://localhost:8000/data | wc -c`. However, I have not been able to reproduce this issue via curl.

I experienced this issue after upgrading a functional application from Symfony 4.1 to 5.1. When the application first loads, no data is present and thus an empty response should be returned. This functioned correctly with Symfony 4.1 but no longer does in Symfony 5.1. The associated JavaScript that made the request began to fail upon checking that the response was or was not empty.

**Comment by deAtog** [comment]
I created a basic app for testing this. When using jQuery.ajax() the Content-Length was always 1. When using the Fetch api the Content-Length was 0. The documentation for the jQuery.ajax() call indicates that the `X-Requested-With` header is always set to `XMLHttpRequest`. After adding this header to the the tests using the Fetch api, the Content-Length became 1. Adding other headers did not seem to alter the Content-Length so the bug appears to be in relation to this header only.

**Comment by deAtog** [comment]
Here's a test project that has a single page that demonstrates the issue: [https://github.com/deAtog/symfony-response-test](https://github.com/deAtog/symfony-response-test)

To run the tests:
1. run `composer install`
2. run `npm install`
3. run `npm run dev`
4. start a dev server using `symfony server:start -d`
5. browse to the main index page.

**Comment by xabbuh** [comment]
Thank you for creating the example application. I debugged it a bit and inside the front controller the response does look as expected. There is no response content and no `Content-Length` header is set. Did you try your application with another web server than the one provided by the Symfony CLI? Do you get the same results with it?

**Comment by deAtog** [comment]
Testing with PHP's built-in server gives the expected results. E.g. Running `php -s localhost:8000` from the `public` directory inside the application folder. Looks like this is an issue with Symfony's built-in web server and not the framework. I have v4.20.5 of the Symfony CLI.

**Comment by derrabus** [comment]
Maybe we should report this issue to https://github.com/symfony/cli then.

**Comment by deAtog** [comment]
@derrabus I agree. Maybe this issue can be moved there? It didn't occur to me that the issue would have been caused by the built-in web server.
